cpp/src/tsne/bh_kernels.cuh

```diff
@@ -173,30 +182,32 @@ __global__ __launch_bounds__(1024, 1) void ClearKernel1(int *restrict childd,
  * Build the actual QuadTree.
  * See: https://iss.oden.utexas.edu/Publications/Papers/burtscher11.pdf
  */
+template <typename value_idx, typename value_t>
 __global__ __launch_bounds__(
-  THREADS2, FACTOR2) void TreeBuildingKernel(/* int *restrict errd, */
-                                             int *restrict childd,
-                                             const float *restrict posxd,
-                                             const float *restrict posyd,
-                                             const int NNODES, const int N,
-                                             int *restrict maxdepthd,
-                                             int *restrict bottomd,
-                                             const float *restrict radiusd) {
-  int j, depth;
-  float x, y, r;
-  float px, py;
-  int ch, n, locked, patch;
+  THREADS2, 2) void TreeBuildingKernel(/* int *restrict errd, */
```

Comment:
Same here- what do you think about removing the second parameter (and the constant)?